plotly/tests/test_core/test_file/test_file.py

```diff
@@ -5,47 +5,47 @@
 A module intended for use with Nose.
 
 """
-
-from nose.tools import raises
-from nose import with_setup
-
 import random
 import string
-import requests
+from unittest import TestCase
 
 import plotly.plotly as py
-import plotly.tools as tls
 from plotly.exceptions import PlotlyRequestError
 
 
-def _random_filename():
-    random_chars = [random.choice(string.ascii_uppercase) for _ in range(5)]
-    unique_filename = 'Valid Folder'+''.join(random_chars)
-    return unique_filename
-
-
-def init():
-    py.sign_in('PythonTest', '9v9f20pext')
-
+class FolderAPITestCase(TestCase):
 
-@with_setup(init)
-def test_create_folder():
-    py.file_ops.mkdirs(_random_filename())
+    def setUp(self):
+        py.sign_in('PythonTest', '9v9f20pext')
 
+    def _random_filename(self):
+        random_chars = [random.choice(string.ascii_uppercase)
+                        for _ in range(5)]
+        unique_filename = 'Valid Folder'+''.join(random_chars)
+        return unique_filename
 
-@with_setup(init)
-def test_create_nested_folders():
-    first_folder = _random_filename()
-    nested_folder = '{0}/{1}'.format(first_folder, _random_filename())
-    py.file_ops.mkdirs(nested_folder)
+    def test_create_folder(self):
+        try:
+            py.file_ops.mkdirs(self._random_filename())
+        except PlotlyRequestError as e:
+            self.fail('Expected this *not* to fail! Status: {}'
+                      .format(e.status_code))
 
+    def test_create_nested_folders(self):
+        first_folder = self._random_filename()
+        nested_folder = '{0}/{1}'.format(first_folder, self._random_filename())
+        try:
+            py.file_ops.mkdirs(nested_folder)
+        except PlotlyRequestError as e:
+            self.fail('Expected this *not* to fail! Status: {}'
+                      .format(e.status_code))
 
-@with_setup(init)
-def test_duplicate_folders():
-    first_folder = _random_filename()
-    py.file_ops.mkdirs(first_folder)
-    try:
+    def test_duplicate_folders(self):
+        first_folder = self._random_filename()
         py.file_ops.mkdirs(first_folder)
-    except PlotlyRequestError as e:
-        if e.status_code != 409:
-            raise e
+        try:
+            py.file_ops.mkdirs(first_folder)
+        except PlotlyRequestError as e:
+            self.assertTrue(400 <= e.status_code < 500)
```

Comment:
yeah, i could definitely see that, it's just that i don't think the python library should couple so closely to streambed. this holds for streaming too, we should try and be as transparent about interacting with the plotly apis as possible. i think :)

Also, the `4xx` is like "hey, you did something wrong, you'll need to change your code to fix it", which is much different from `5xx`, "oops, our bad, you might want to try that again later".